lib/runtime/storage/trie.go

```diff
@@ -178,6 +180,43 @@ func (s *TrieState) DeleteChild(key []byte) {
 	s.t.DeleteChild(key)
 }
 
+// DeleteChildLimit deletes up to limit of database entries by lexicographic order, return number
+//  deleted, true if all delete atherwise false
+func (s *TrieState) DeleteChildLimit(key []byte, limit *optional.Bytes) (uint32, bool, error) {
```

Comment:
I don't think this is correct, as per substrate this deletes up to `limit` keys from the child trie which is located at `key`: https://github.com/paritytech/substrate/blob/4c3a55e7ca5c4c85c1eb53fd82ed71029d952510/primitives/externalities/src/lib.rs#L131 so, you'll need to get a list of all the keys from the child trie at `key` then delete up to `limit` of then, and return true if the entire child was deleted, false otherwise

Reply:
I don't understand which part if this is not correct?  First I'm checking if limit is set, then deleting all if not set.  Otherwise I'm using the limit, then getting a list of keys and deleting up up to limit.  Am I missing something between DeleteFromChild and DeleteChild?  

These tests: `Test_ext_default_child_storage_storage_kill_version_2_limit_1` and `Test_ext_default_child_storage_storage_kill_version_2_limit_none` are giving me expected results, did I set these tests up incorrectly? Or are this passing because I've miss-understood the spec?  
```
	tr, err := s.t.GetChild(key)
	if err != nil {
		return 0, false, err
	}
	qtyEntries := uint32(len(tr.Entries()))
	if limit == nil || !limit.Exists() {  // if limit nil or limit !exists then delete all entries at that key
		s.t.DeleteChild(key)
		return qtyEntries, true, nil
	}
	limitUint := binary.LittleEndian.Uint32(limit.Value())

        // limit not nil, so make list of keys (and sort)
	keys := make([]string, 0, len(tr.Entries()))
	for k := range tr.Entries() {
		keys = append(keys, k)
	}
	sort.Strings(keys)
	deleted := uint32(0)
        // delete up to limit quantity
	for _, k := range keys {
		tr.Delete([]byte(k))
		deleted++
		if deleted == limitUint {
			break
		}
	}

	if deleted == qtyEntries {
		return deleted, true, nil
	}

	return deleted, false, nil

```

Reply:
ok so sorry, I totally missed that you got `tr.Entries()` in a following line. ignore this comment